netlogo-gui/src/main/ide/ShowUsageBox.scala

```diff
@@ -0,0 +1,133 @@
+    }
+
+    override def windowGainedFocus(e: WindowEvent): Unit = {}
+  })
+  dataModel.setColumnCount(2)
+  usageTable.setSelectionMode(javax.swing.ListSelectionModel.SINGLE_SELECTION)
+  usageTable.getColumnModel.getColumn(0).setMaxWidth(40)
+  usageTable.getColumnModel.setColumnSelectionAllowed(false)
+  usageTable.getTableHeader().setReorderingAllowed(false)
+  usageTable.setTableHeader(null)
+  usageTable.setRowMargin(3)
+  usageTable.setRowHeight(20)
+  usageTable.getColumnModel.getColumn(1).setMinWidth(200)
+  usageTable.getColumnModel.getColumn(0).setMinWidth(40)
+  usageTable.setFont(editorArea.getFont)
+
+  usageTable.addMouseListener(new MouseAdapter() {
+    override def mouseClicked(e: MouseEvent): Unit = {
+      usageBox.setVisible(false)
+      val token = usageTable.getValueAt(usageTable.getSelectedRow, 0).asInstanceOf[Token]
+      dataModel.synchronized(dataModel.setRowCount(0))
+      editorArea.setCaretPosition(token.start)
+    }
+  })
+
+  usageTable.addKeyListener(new KeyAdapter(){
+    override def keyPressed(e: KeyEvent): Unit = {
+      e.getKeyCode match {
+        case KeyEvent.VK_ESCAPE => usageBox.setVisible(false)
+        case _ => e.setSource(editorArea)
+          editorArea.dispatchEvent(e)
+      }
+    }
+  })
+
+  usageTable.setCursor(Cursor.getPredefinedCursor(Cursor.HAND_CURSOR))
+
+  def showBox(me: MouseEvent, position: Int): Unit = {
+    val tokenOption = findTokenContainingPosition(editorArea.getText(), position)
+    for {token <- tokenOption} {
+      if(token.tpe == TokenType.Ident || token.tpe == TokenType.Command || token.tpe == TokenType.Reporter) {
+        val tokens = getUsage(editorArea.getText(), token)
+        dataModel.setRowCount(0)
+        for (t <- tokens) {
+          dataModel.addRow(Array[AnyRef](t, editorArea.getLineText(t.start).trim))
+        }
+        if (dataModel.getRowCount != 0) {
+          usageTable.setPreferredScrollableViewportSize(usageTable.getPreferredSize())
+          usageTable.setFillsViewportHeight(true)
+          usageBox.setSize(usageTable.getPreferredSize)
+          usageTable.validate()
+          usageBox.setLocation(me.getLocationOnScreen)
+          usageBox.setVisible(true)
+        }
+      }
+    }
+  }
+
+  def getUsage(source: String, token: Token): Iterator[Token] = {
+    val iterator = Femto.scalaSingleton[TokenizerInterface]("org.nlogo.lex.Tokenizer").tokenizeString(source)
+    iterator.filter(t => t.text.equalsIgnoreCase(token.text) && t.start != token.start)
+  }
+
+  def findTokenContainingPosition(source: String, position: Int): Option[Token] = {
+    val iterator = Femto.scalaSingleton[TokenizerInterface]("org.nlogo.lex.Tokenizer").tokenizeString(source)
+    iterator.find(p => p.start < position && p.end >= position)
+  }
+
+  class LineRenderer extends TableCellRenderer {
+    override def getTableCellRendererComponent(table: JTable, value: scala.Any, isSelected: Boolean, hasFocus: Boolean, row: Int, column: Int): Component = {
```

Comment:
You can just use `Any` instead of `scala.Any`. (and maybe `AnyRef` is more appropriate here? I guess this is `Object` in the Java as it's Swing API)